### PR TITLE
Improve the VS Code Debugging Page with Relevant Links

### LINKED ISSUE #355: Improve debugging page
[Debugging a Robot Program](http://frcdocs.wpi.edu/en/latest/docs/software/wpilib-overview/debugging-robot-program.html) talks about setting a breakpoint, but not single stepping.

### PR DESCRIPTION
**Pull request by jasondaming**
Closes #355

### CODE REVIEW

**Comment by jasondaming** [comment]
Some structure questions I have:
- I think this page (https://code.visualstudio.com/docs/editor/editingevolved) is also really good for getting started coding in VS Code but I am unsure where it would go.  I guess in this (https://docs.wpilib.org/en/latest/docs/software/vscode-overview/vscode-basics.html) basics article, but maybe it is more advanced than that?
- Where do you think the OutlineViewer page should go?  Under Software Tools would make sense but it isn't nearly as big as all the others there.  It could go under Network Tables?
- Are there other links or critical debugging info that you think I should include here?

**Comment by Daltz333** [comment]
I would keep the note that a debugger is recommended. We shouldn't recommend prints. An accidental print in Java robot code can cause some pretty severe loop overruns

**Comment by jasondaming** [comment]
are you talking about the note in the blue box?  It felt repetitive with what I had under Debugging with Print Statements.  I have no problem adding it back if that is what you meant as it is an important point to ram home

**Comment by jasondaming** [comment]
Who knows the most about OutlineViewer?  There are some featuers I am unsure what they do and I have never actually used it.  Is it okay to leave the addition of OutlineViewer in this PR?

File: source/docs/software/vscode-overview/debugging-robot-program.rst [hunk 1]
```diff
@@ -5,72 +5,54 @@ Inevitably, a program will not behave in the way we expect it to behave.  When t
...
 
-|Watch Pane|
+        System.out.print("example");
 
-The Watch Pane can be used to monitor specific variables or expressions while debugging. To add an expression, right-click and select Add Expression.
+    .. code-tab:: c++
 
-Call Stack
-^^^^^^^^^^
+        fmt::print("example");
```
Comment (prateekma) [inline source/docs/software/vscode-overview/debugging-robot-program.rst:42]: This is not going to compile unless teams add `fmtlib` as a dependency. We should recommend `wpi::outs() << "example\n";` instead (from the `wpi/raw_ostream` header).

**Review by prateekma — CHANGES_REQUESTED** [review]

File: source/docs/software/vscode-overview/debugging-robot-program.rst (hunk 1 above)
Comment (jasondaming) [reply source/docs/software/vscode-overview/debugging-robot-program.rst]: blame @calcmogul I got that from him.  Fixed

Comment (calcmogul) [reply source/docs/software/vscode-overview/debugging-robot-program.rst]: I didn't intend for it to be in the docs. :stuck_out_tongue:

**Comment by Daltz333** [comment]
The OutlineViewer should be a seperate article.

And yeah, I'd leave the note. People don't like to read things that aren't screaming at them.

**Comment by Daltz333** [comment]
Ideally, I'd like to refactor this page a bit more. Can we use the 
```
:kbd:`Ctrl+Shift+P`
```
directive for keyboard shortcuts and migrate to inline image tags

IE: Replace
```
| my image |
```

with

```
.. image:: myimage.png
   :alt: What my image shows
```

Additionally, things like GUI text elements should be highlighted with the ``guilabel`` directive

```
:guilabel:`my label`
```

**Comment by jasondaming** [comment]
With :kdb: should it be Ctrl or Control?  No spaces?  + or -?  I am going to clean up other similar issues.

**Comment by Daltz333** [comment]
Control is Mac, Ctrl is Windows/Linux

**Comment by jasondaming** [comment]
What about spaces and +/-?  I wasn't familiar with :kdb: should we add that to the style doc?

**Comment by jasondaming** [comment]
What do you think TOC should be?  I like moving 3rd party up to after creating and before deploying (because you need to know to have them before you deploy).

I am not sure what else I would move above debugging both of the importing only needs to be done once so it should either go to the top or where it is and using test mode I think is in the wrong section.  It should probably go to the driverstation section or something.  It really has nothing to do with VS Code Overview

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**